### ISSUE TITLE
When receiving a message and switching to the Telegram tab, the conversation is not marked as read on Firefox

### ISSUE THREAD

**Issue by cypressious** [zhukov/webogram#1811]
This is a Firefox specific bug.

When a message is receiver in the currently opened conversation while the tab is inactive and switching to the tab, the conversation often gets stuck in the unread state.

Here's a GIF in Firefox:

![telegram_ff](https://user-images.githubusercontent.com/1191816/52640955-131d1f00-2ed8-11e9-9641-3a8dbf2ccf04.gif)

Compare to Chrome:

![telegram_chrome](https://user-images.githubusercontent.com/1191816/52640954-131d1f00-2ed8-11e9-847a-a184791b5228.gif)

I've also opened an [issue on the Firefox Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1525437) but it's currently unclear if it's a browser bug.

**Comment by emilio** [comment]
Yeah, I'm happy to investigate if there's some disagreement between browsers that you think is a bug on our side. Thanks in advance!

**Comment by achirasa** [comment]
ใช่คุณรึป่าวที่ช่วยผม มาตลอด ผมไม่ค่อยเข้าใจวิธีการ ผมต้องกราบขออภัย  iam
sory very very และฝากขอโทษเพื่อนๆทุก ผมไม่กล้าตอบ พยายามเรี่ยงที่จะคุย
เพราะตัวเองโง่ ทำได้บ้างไม่ได้บ้าง แล้วพอไม่ไหวก็ หายไปเลย ไปโดยไม่ได้ร่ำลา
ไม่ได้บอก ผมขอโทษมากเลยนะคับ ผมแย่ที่สุดแย่มากๆ
ในใจมันสับสนทั้งกล้าทั้งกลัว
แต่พอจะทำจริงๆแล้วผมกับกลัวผลรับที่จะตามมาถ้าดีก็ดีไป
ถ้าไม่ใช่ผมกลัวทุกคนที่บ้าน และกลัวลูกจะเดือดร้อนไปด้วย
ยังไงผมก็ขอขอบพระคุณมากๆนะคับ ไม่รู้ผมพูดจะใช่คุณาป่าว แต่ขอของคุณ
และทุกๆท่านด้วย ผมจะยังไม่ไปไหน ผมจะเข้ามาขอความรู้ อีกนะคับ ขอบคุณครับ

เมื่อ 12 ก.พ. 2019 9:10 PM "Kirill Rakhman" <notifications@github.com>
เขียนว่า

**Comment by bladegery** [comment]
Yeah, this is so annoying, I switch to the tab and keep staring at the message I have already read in split second. I start to click and scroll around then it finally disappears.

**Comment by Aokromes** [comment]
@emilio it's firefox bug or telegram bug?

**Comment by emilio** [comment]
I'm not sure off-hand. If somebody familiar with the webogram code can point me to how that notification is marked as read, or how is Firefox diverging from Chrome or other browsers, I could find out if the bug is in FF or not.
